sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -24,13 +24,21 @@
 from __future__ import absolute_import
 
 import apache_beam as beam
+from apache_beam.io.gcp.pubsub import ReadFromPubSub
 from apache_beam.pipeline import PipelineVisitor
 from apache_beam.runners.interactive import cache_manager as cache
 from apache_beam.runners.interactive import interactive_environment as ie
 
 READ_CACHE = "_ReadCache_"
 WRITE_CACHE = "_WriteCache_"
 
+# Use a tuple to define the list of unbounded sources. It is not always feasible
+# to correctly find all the unbounded sources in the SDF world. This is
+# because SDF allows the source to dynamically create sources at runtime.
```

Comment:
I'm not sure what you mean here since you can detect which ParDo's contain splittable dofns as shown here:
https://github.com/apache/beam/blob/2f0061360ccc820fd8865ea63cf3224eb9d47397/sdks/python/apache_beam/transforms/core.py#L1278

Reply:
Spoke in person. The detection for splittable dofns will be done later.